File: gui/src/shared/models/kg/search/KgSearchResult.ts
export type KgSearchResult =
  | {
      __typename: "KgEdgeLabelSearchResult";
      edgeLabel: string;
      sourceIds: readonly string[];
    }
  | {
      __typename: "KgEdgeSearchResult";
      edge: {
        id: string;
        labels: string[];
        sourceIds: readonly string[];
      };
    }
  | {
      __typename: "KgNodeLabelSearchResult";
      nodeLabel: string;
      sourceIds: readonly string[];
    }
  | {
      __typename: "KgNodeSearchResult";
      node: {
        id: string;
        labels: string[];
        pos: string | null;
        sourceIds: readonly string[];
      };
    }
  | {
      __typename: "KgSourceSearchResult";
      sourceId: string;
    };

File: lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/models/search/KgSearchResult.scala
package io.github.tetherlessworld.mcsapps.lib.kg.models.search

import io.github.tetherlessworld.mcsapps.lib.kg.models.kg.KgEdge
import io.github.tetherlessworld.mcsapps.lib.kg.models.node.KgNode

sealed trait KgSearchResult

final case class KgEdgeSearchResult(edge: KgEdge) extends KgSearchResult

final case class KgEdgeLabelSearchResult(edgeLabel: String, sourceIds: List[String]) extends KgSearchResult

final case class KgNodeLabelSearchResult(nodeLabel: String, sourceIds: List[String]) extends KgSearchResult

final case class KgNodeSearchResult(node: KgNode) extends KgSearchResult

final case class KgSourceSearchResult(sourceId: String) extends KgSearchResult
